src/main/scala/viper/gobra/reporting/StatsCollector.scala

```diff
@@ -0,0 +1,357 @@
+//
+// Copyright (c) 2011-2020 ETH Zurich.
+
+package viper.gobra.reporting
+
+import org.apache.commons.io.FileUtils
+import org.bitbucket.inkytonik.kiama.relation.NodeNotInTreeException
+import viper.gobra.ast.frontend.{PDomainFunction, PExpression, PFPredicateDecl, PFunctionDecl, PMPredicateDecl, PMPredicateSig, PMethodDecl, PMethodImplementationProof, PMethodSig, PNode, PParameter, PPredConstructor}
+import viper.gobra.ast.internal.BuiltInMember
+import viper.gobra.frontend.{Config, PackageEntry}
+import viper.gobra.frontend.info.{Info, TypeInfo}
+import viper.gobra.util.Violation
+import viper.silver.ast.{FuncApp, Function, Member, Method, MethodCall, Node, Predicate, PredicateAccess}
+import viper.silver.reporter.Time
+
+import java.nio.charset.StandardCharsets.UTF_8
+import java.io.File
+
+case class StatsCollector(reporter: GobraReporter) extends GobraReporter {
+  // Stores type info for a task
+  private[reporting] var typeInfos: Map[String, TypeInfo] = Map()
+  // Maps a gobra member name to a gobra member entry
+  private[reporting] var memberMap: Map[String, GobraMemberEntry] = Map()
+  // Maps a viper member name to a gobra member entry
+  private var viperMemberNameGobraMemberMap: Map[String, GobraMemberEntry] = Map()
+
+  case class GobraMemberEntry(pkgDir: String,
+                              pkg: String,
+                              memberName: String,
+                              nodeType: String,
+                              args: String,
+                              viperMembers: Map[String, ViperMemberEntry],
+                              isTrusted: Boolean,
+                              isAbstract: Boolean) {
+
+    def asJson(p: String = ""): String = {
+      val viperMembersJson = viperMembers.values.map(_.asJson(s"$p    "))
+      val dependencies = this.dependencies().map(entry => p + "    \"" + entry.key + "\"")
+
+      s"""$p{
+         |$p  "id": "${this.key}",
+         |$p  "pkgDir": "$pkgDir",
+         |$p  "pkg": "$pkg",
+         |$p  "name": "$pkg.$memberName",
+         |$p  "args": "$args",
+         |$p  "nodeType": "$nodeType",
+         |$p  "trusted": $isTrusted,
+         |$p  "abstract": $isAbstract,
+         |$p  "viperMembers": [
+                |${viperMembersJson.mkString(",\n")}
+         |$p  ],
+         |$p  "dependencies": [
+                |${dependencies.mkString(",\n")}
+         |$p  ]
+         |$p}""".stripMargin
+    }
+
+    def dependencies(): Set[GobraMemberEntry] = {
+      this.viperMembers.values
+        .flatMap(viperMember => viperMember.dependencies
+          .flatMap(dep => findGobraMemberByViperMemberName(viperMember.taskName, dep, pkgDir)))
+        .filter(_.key != this.key).toSet
+    }
+
+    def key: String = gobraMemberKey(pkgDir, pkg, memberName, args)
+  }
+
+  case class ViperMemberEntry(memberName: String,
+                              taskName: String,
+                              time: Time,
+                              nodeType: String,
```

Comment:
If this is supposed to have only one value out of a small and well-defined set of values, I would not make this a string. Instead, I would use an Enum or a new trait